txscript/opcode.go

```diff
@@ -1953,6 +1953,12 @@ func opcodeCodeSeparator(op *opcode, data []byte, vm *Engine) error {
 
 	if vm.taprootCtx != nil {
 		vm.taprootCtx.codeSepPos = uint32(vm.tokenizer.OpcodePosition())
+	} else if !vm.isWitnessVersionActive(0) &&
```

Comment:
I think we can tighten this policy check a bit. The `bitcoind` version disables for all segwit versions, while this would only disable for segwit v0. Instead we can check that `vm.witnessProgram == nil`.